File: rd-net/Cross/Common/Util/FileSystem.cs
using System.IO;

namespace Test.RdCross.Util
{
    public static class FileSystem
    {
        public static string RdTmpDir => Path.Combine(Path.GetTempPath(), "rd");

        internal static string PortFile => Path.Combine(RdTmpDir, "port.txt");
    }
}

File: rd-kt/rd-cross/src/main/kotlin/com/jetbrains/rd/cross/util/FileSystem.kt
package com.jetbrains.rd.cross.util

import java.io.File

val rdTmpDir: File = File(System.getProperty("java.io.tmpdir"), "rd")
    get() {
        field.mkdirs()
        return field
    }

val portFile = File(rdTmpDir, "port.txt")
    get() {
        rdTmpDir.mkdirs()
        return field
    }

val portFileStamp = portFile.resolveSibling("port.txt.stamp")